services/diddoc/queries/resources/resource_query_metadata_handler.go

```diff
@@ -19,7 +19,14 @@ func (d *ResourceMetadataHandler) Handle(c services.ResolverContext, service ser
 	if err != nil {
 		return nil, err
 	}
+	acceptHeader := c.Request().Header.Get(echo.HeaderAccept)
```

Comment:
as d.IsDereferencing is always true due to your changes in this PR.

W3DIDRES returns a dereferencing result

Reply:
@DaevMithran This check is needed to handle the special case of resourceMetadata